### PR TITLE
[comm_mode] adding some initial c10d ops to CommDebugMode

### PR DESCRIPTION
**Pull request by wanchaol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #125476
* __->__ #125475

looks like we can make it work :)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

**Review by awgu — APPROVED** [review]
Amazing!

File: test/distributed/fsdp/test_fsdp_tp_integration.py [hunk 1]
```diff
@@ -387,11 +387,16 @@ def forward(self, x):
...
         comm_counts = comm_mode.get_comm_counts()
-        self.assertEqual(comm_mode.get_total_counts(), 5)
+        self.assertEqual(comm_mode.get_total_counts(), 7)
+        # TP comms
         self.assertEqual(comm_counts[funcol.reduce_scatter_tensor], 2)
         self.assertEqual(comm_counts[funcol.all_gather_into_tensor], 2)
         self.assertEqual(comm_counts[funcol.all_reduce], 1)
+        # FSDP comms
+        self.assertEqual(comm_counts[c10d_ops._allgather_base_], 1)
+        self.assertEqual(comm_counts[c10d_ops._reduce_scatter_base_], 1)
```
Comment (wanchaol) [inline test/distributed/fsdp/test_fsdp_tp_integration.py:399]: Now we can capture FSDP comms (although it triggered some failures in CI but this is nice to see!)

**Comment by XilunWu** [comment]
Hi Wanchao. I'm planning to add all c10d collective ops to CommDebugMode. WDYT (worth to do? Difficulty?). cc @awgu

**Comment by wanchaol** [comment]
> Hi Wanchao. I'm planning to add all c10d collective ops to CommDebugMode. WDYT (worth to do? Difficulty?). cc @awgu

@XilunWu yeah feel free to do it :) I think it's not that difficult, one just need to add the ops, and add tests.

**Pull request closed without merging**